sql/mysql_db/mysql_db.go

```diff
@@ -367,10 +367,19 @@ func (db *MySQLDb) GetTableNames(ctx *sql.Context) ([]string, error) {
 
 // AuthMethod implements the interface mysql.AuthServer.
 func (db *MySQLDb) AuthMethod(user, addr string) (string, error) {
-	host, _, err := net.SplitHostPort(addr)
-	if err != nil {
-		return "", err
+	var host string
+	// TODO : need to check for network type instead of addr string if it's unix socket network,
+	//  macOS passes empty addr, but ubuntu returns "@" as addr for `localhost`
+	if addr == "@" || addr == "" {
```

Comment:
I think this is okay for now. It would be cleaner to change the mysql.AuthServer interface to take in network type, like you mentioned, but I don't see any large risk from this code that just adds two special cases for localhost.